app/questionnaire/router.py

```diff
@@ -71,36 +71,25 @@ def get_next_location_url(self, location, routing_path):
         Get the first incomplete block in section/survey if trying to access the section/survey end,
         and the section/survey is incomplete or gets the next default location if the above is false.
         """
-        current_block_type = self._schema.get_block(location.block_id)["type"]
         last_block_id = routing_path[-1]
-        last_block_type = self._schema.get_block(last_block_id)["type"]
         hub_enabled = self._schema.is_hub_enabled()
+        has_section_summary = self._schema.is_summary_in_section(location.section_id)
 
         if (
-            hub_enabled
-            and location.block_id == last_block_id
+            last_block_id == location.block_id
             and self._progress_store.is_section_complete(
                 location.section_id, location.list_item_id
             )
         ):
-            return url_for(".get_questionnaire")
 
-        # If the section is complete and contains a SectionSummary, return the SectionSummary location
-        if (
-            last_block_type == "SectionSummary"
-            and current_block_type != last_block_type
-            and self._progress_store.is_section_complete(
-                location.section_id, location.list_item_id
-            )
-        ):
-            return url_for(
-                "questionnaire.block",
-                block_id=last_block_id,
-                list_name=routing_path.list_name,
-                list_item_id=routing_path.list_item_id,
-            )
+            if has_section_summary:
+                return self.get_section_summary_url(location)
+            if hub_enabled:
+                return url_for(".get_questionnaire")
 
         if self.is_survey_complete() and not hub_enabled:
+            if has_section_summary:
+                return self.get_section_summary_url(location)
```

Comment:
I ummed and ahed on this for a bit as I was also unsure, but there was one functional test which breaks without it 'question_variants.spec.js'. Basically if you complete a non-hub survey, but then go back and change an answer, the old behaviour was to return to the section summary. As the card said to match current behaviour I put it in, but naturally it can be remove if its now not right

Reply:
Now return_to_summary is on Final summary, the above has been absorbed into the proceeding condition, so the flag can correctly be used to hand back to the final summary if it doesn't have a section summary but only if the survey is complete.